apps/lib/cmp_mock_srv.c

```diff
@@ -241,6 +241,36 @@ static OSSL_CMP_PKISI *process_cert_request(OSSL_CMP_SRV_CTX *srv_ctx,
         /* give final response after polling */
         ctx->curr_pollCount = 0;
 
+    /* accept cert profile for cr messages only with the configured name */
+    if (OSSL_CMP_MSG_get_bodytype(cert_req) == OSSL_CMP_CR) {
+        STACK_OF(OSSL_CMP_ITAV) *itavs =
+            OSSL_CMP_HDR_get0_geninfo_ITAVs(OSSL_CMP_MSG_get0_header(cert_req));
+        int i;
+
+        for (i = 0; i < sk_OSSL_CMP_ITAV_num(itavs); i++) {
```

Comment:
Is this bad if itavs is NULL? i.e. sk_*_num returns -1.

Reply:
The case `itavs == NULL` is not a problem - 
both conceptually and for the given `for(...)` loop, having no ITAVs is equivalent to having a empty list of them.